perf(traces-table): improve performance

marliessophie opened the pull request:
<!-- ELLIPSIS_HIDDEN -->


> [!IMPORTANT]
> Optimize traces table performance by memoizing components and reducing unnecessary re-renders and network requests.
> 
>   - **Performance Improvements**:
>     - Memoize `TableBodyComponent` as `MemoizedTableBody` in `data-table.tsx` to reduce re-renders during column resizing and peek view navigation.
>     - Use `memo` for `IOTableCell` as `MemoizedIOTableCell` in `CodeJsonViewer.tsx` to optimize rendering of large JSON data.
>     - Set `staleTime` to 1 minute for data fetching hooks in `usePeekData.ts`, `usePeekView.ts`, and `DatasetRunItemsTable.tsx` to reduce network requests.
>   - **Behavior Changes**:
>     - Remove `SelectedRowContext` from `data-table.tsx` and use `useRouter` for row selection state.
>     - Simplify `handleOnRowClick` logic in `data-table.tsx` by removing `inflatedPeekView` dependency.
>     - Update `TablePeekView` in `peek.tsx` to use `memo` for performance and refactor props handling.
>   - **Miscellaneous**:
>     - Refactor `usePeekView` in `usePeekView.ts` to improve row state management and URL handling.
>     - Adjust `GenerationsDynamicCell` and `TracesDynamicCell` in `observations.tsx` and `traces.tsx` to use `MemoizedIOTableCell`.
>     - Update `DatasetRunItemsTable.tsx` to use `MemoizedIOTableCell` for `TraceObservationIOCell` and `DatasetItemIOCell`.
> 
> <sup>This description was created by </sup>[<img alt="Ellipsis" src="https://img.shields.io/badge/Ellipsis-blue?color=175173">](https://www.ellipsis.dev?ref=langfuse%2Flangfuse&utm_source=github&utm_medium=referral)<sup> for f0ec3733d6cc608c783cbfdb2aea9e0018e738fe. You can [customize](https://app.ellipsis.dev/langfuse/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>


<!-- ELLIPSIS_HIDDEN -->

marliessophie committed:
fixup: limit render duration of traces table

marliessophie committed:
fixup: reduce render cycles through data flows and memoization

maxdeichmann commented on web/src/components/table/use-cases/traces.tsx:
```diff
@@ -1206,27 +1220,29 @@ const TracesDynamicCell = ({
+      refetchOnMount: false,
+      refetchOnWindowFocus: false,
+      refetchOnReconnect: false,
+      staleTime: 5 * 60 * 1000,
```
I think that only setting staleTime here is already enough but also not sure.

marliessophie commented on web/src/components/table/use-cases/traces.tsx:
```diff
@@ -1206,27 +1220,29 @@ const TracesDynamicCell = ({
-      enabled: typeof traceId === "string",
-      trpc: {
-        context: {
-          skipBatch: true,
```
we set skip batch globally, there is no point in setting it here, see https://github.com/langfuse/langfuse/pull/2999

marliessophie committed:
refactor: enhance performance and structure of table components

- Removed unnecessary console logs in `data-table.tsx`.
- Refactored `TablePeekView` to use a more structured props type and memoization for improved performance.
- Updated `usePeekData` to reduce stale time from 5 minutes to 1 minute.
- Replaced `IOTableCell` with `MemoizedIOTableCell` in various components to optimize rendering.
- Adjusted `peekView` configurations in `observations.tsx` and `traces.tsx` for consistency and clarity.

marliessophie committed:
chore: eslint
